internal/api/handler/oscal/assessmentplan.go

```diff
@@ -0,0 +1,601 @@
+	} else if _, err := uuid.Parse(plan.UUID); err != nil {
+		errors = append(errors, fmt.Sprintf("invalid UUID format: %s", plan.UUID))
+	}
+
+	if plan.Metadata.Title == "" {
+		errors = append(errors, "metadata.title is required")
+	}
+
+	if plan.Metadata.Version == "" {
+		errors = append(errors, "metadata.version is required")
+	}
+
+	if plan.ImportSsp.Href == "" {
+		errors = append(errors, "import-ssp.href is required")
+	}
+
+	if len(errors) > 0 {
+		return fmt.Errorf("validation errors: %s", strings.Join(errors, "; "))
+	}
+
+	return nil
+}
+
+// buildQueryWithExpansion builds a GORM query with appropriate preloads based on expansion parameters
+func (h *AssessmentPlanHandler) buildQueryWithExpansion(baseQuery *gorm.DB, expand, include string) *gorm.DB {
+	// Always include basic metadata
+	query := baseQuery.
+		Preload("Metadata").
+		Preload("Metadata.Revisions")
+
+	// Handle full expansion
+	if expand == "all" || expand == "full" {
+		return h.addAllPreloads(query)
+	}
+
+	// Handle selective inclusion
+	if include != "" {
+		return h.addSelectivePreloads(query, include)
+	}
+
+	return query
+}
+
+// addAllPreloads adds all available preloads for complete resource expansion
+func (h *AssessmentPlanHandler) addAllPreloads(query *gorm.DB) *gorm.DB {
+	return query.
+		Preload("Tasks").
+		Preload("Tasks.Dependencies").
+		Preload("Tasks.Tasks"). // Sub-tasks
+		Preload("Tasks.AssociatedActivities").
+		Preload("Tasks.AssociatedActivities.Steps").
+		Preload("Tasks.Subjects").
+		Preload("Tasks.ResponsibleRole").
+		Preload("AssessmentAssets").
+		Preload("AssessmentSubjects").
+		Preload("LocalDefinitions").
+		Preload("TermsAndConditions").
+		Preload("BackMatter")
+}
+
+// addSelectivePreloads adds specific preloads based on the include parameter
+func (h *AssessmentPlanHandler) addSelectivePreloads(query *gorm.DB, include string) *gorm.DB {
+	includes := strings.Split(include, ",")
+	for _, inc := range includes {
+		switch strings.TrimSpace(inc) {
+		case "tasks":
+			query = query.Preload("Tasks").
+				Preload("Tasks.Dependencies").
+				Preload("Tasks.Tasks")
+		case "activities":
+			query = query.Preload("Tasks").
+				Preload("Tasks.AssociatedActivities").
+				Preload("Tasks.AssociatedActivities.Steps")
+		case "assets", "assessment-assets":
+			query = query.Preload("AssessmentAssets")
+		case "subjects", "assessment-subjects":
+			query = query.Preload("AssessmentSubjects")
+		case "local-definitions":
+			query = query.Preload("LocalDefinitions")
+		case "terms-conditions", "terms-and-conditions":
+			query = query.Preload("TermsAndConditions")
+		case "back-matter":
+			query = query.Preload("BackMatter")
+		}
+	}
+	return query
+}
+
+// List godoc
+//
+//	@Summary		List Assessment Plans
+//	@Description	Retrieves all Assessment Plans with optional pagination and expansion.
+//	@Tags			Assessment Plans
+//	@Produce		json
+//	@Param			page	query		int		false	"Page number (default: 1)"
```

Comment:
We are not including pagination at all yet, as there will only be 1 assessment plan per CCF installation at this point.